File: packages/sdui/src/main/java/nl/q42/sdui/SDUIApplication.java
package nl.q42.sdui;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Singular;
import nl.q42.common.MetadataType;
import nl.q42.common.actions.Action;
import nl.q42.common.analytics.AnalyticEvent;
import nl.q42.common.components.Component;
import nl.q42.sdui.screen.common.ScreenTab;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
public class SDUIApplication implements Serializable
{
  public static final int MINIMUM_APP_VERSION = 1;
  public static final int MAXIMUM_APP_VERSION = 1;

  public @Singular List<Action>        onLoadActions;
  public @Singular List<AnalyticEvent> onLoadAnalyticEvents;

  public final Component                 screen;
  public       ScreenTab[]               tabs;
  public       Map<MetadataType, Object> metadata;
}

File: packages/sdui/src/main/kotlin/nl/q42/sdui/SDUIApplication.kt
package nl.q42.sdui

import com.fasterxml.jackson.annotation.JsonInclude
import kotlinx.serialization.Serializable
import nl.q42.common.MetadataType
import nl.q42.common.actions.Action
import nl.q42.common.analytics.AnalyticEvent
import nl.q42.common.screen.Screen
import nl.q42.common.screen.ScreenTab

@JsonInclude(JsonInclude.Include.NON_NULL)
@Serializable
class SDUIApplication(
    val screen: Screen,
    var onLoadActions: MutableList<Action?>? = null,
    var onLoadAnalyticEvents: MutableList<AnalyticEvent?>? = null,
    var tabs: List<ScreenTab>? = null,
    var metadata: MutableMap<MetadataType, Any>? = null,
) {
    companion object {
        const val MINIMUM_APP_VERSION: Int = 1
        const val MAXIMUM_APP_VERSION: Int = 1
    }
}
